apps/sms/views/conversation/js/recipients.js

```diff
@@ -20,7 +20,7 @@
     var number;
 
     opts = opts || {};
-    this.name = opts.name || opts.number || '';
+    this.name = String(opts.name || opts.number || '');
```

Comment:
Some unit tests pass Number instead of String to the Recipient `update` method which in turns use the Recipient constructor.

https://github.com/mozilla-b2g/gaia/blob/41e13d5dd4788a69c28d2696cc60b2b7455e4369/apps/sms/views/conversation/test/unit/recipients_test.js#L272-L285

Maybe you would prefer to fix the tests ?